packages/fxa-payments-server/server/lib/flow-performance.js

```diff
@@ -0,0 +1,109 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+'use strict';
+
+const os = require('os');
+const config = require('../config');
+const log = require('./logging/log')();
+const {
+  VERSION,
+  PERFORMANCE_TIMINGS,
+  limitLength,
+  isValidTime,
+} = require('../../../fxa-shared/metrics/flow-performance.js');
+const geodbConfig = config.get('geodb');
+const geodb = require('../../../fxa-geodb/lib/fxa-geodb.js')(geodbConfig);
+const remoteAddress = require('../../../fxa-shared/express/remote-address.js')(
+  config.get('clientAddressDepth')
+);
+const geolocate = require('../../../fxa-shared/express/geo-locate.js')(geodb)(
+  remoteAddress
+)(log);
+
+const HOSTNAME = os.hostname();
+const FLOW_ID_EXPIRY = config.get('flow_id_expiry');
+
+module.exports = (event, request, data) => {
+  const { view, perfStartTime, flowBeginTime, flowId, navigationTiming } = data;
+
+  if (
+    !view ||
+    !perfStartTime ||
+    !flowBeginTime ||
+    !flowId ||
+    !navigationTiming
+  ) {
+    return;
+  }
+
+  // Since the flow originates in the content-server, the flow begin time should
+  // be before the perf start time.
+  if (perfStartTime < flowBeginTime) {
+    return;
```

Comment:
It seems like we could silently lose perf metrics data here. Should we log something in this case?

Reply:
Since we don't validate the flow params, `flowBeginTime` is ultimately untrustworthy.  And, it is not used in calculating the perf stats.  I think I could just remove this.